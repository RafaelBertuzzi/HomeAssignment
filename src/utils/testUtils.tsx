import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
