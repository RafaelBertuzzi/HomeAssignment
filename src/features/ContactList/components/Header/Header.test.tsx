import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Header } from '.';

describe('ContactList Component: Header', () => {
  afterEach(cleanup);

  it('should render ContactList Header', () => {
    const json = rendererWithProviders(<Header />);
    expect(json).toMatchSnapshot();
  });
});
