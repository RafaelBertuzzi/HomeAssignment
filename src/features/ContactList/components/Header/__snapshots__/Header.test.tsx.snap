// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList Component: Header should render ContactList Header 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "position": "fixed",
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 16,
          "marginRight": 16,
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Contact List
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 10,
          }
        }
      >
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={{}}
            style={{}}
          >
            <Image
              accessibilityLabel="search"
              alt="search"
              dataSet={{}}
              onError={[Function]}
              source={
                {
                  "testUri": "../../../src/assets/icons/search.png",
                }
              }
              style={
                {
                  "height": 24,
                  "maxWidth": "100%",
                  "width": 24,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      aria-orientation="horizontal"
      dataSet={{}}
      style={
        {
          "backgroundColor": "#d4d4d4",
          "height": 1,
          "width": "100%",
        }
      }
      thickness="1"
    />
  </View>
</RNCSafeAreaProvider>
`;
